Assignment3/Q1.java

```diff
@@ -0,0 +1,148 @@
+import java.util.*;
+
+class Q1
+{
+	public static void main(String args[])
+	{
+		char[][] grid = new char[2][3];
+		grid[0][0] = 'A';
+		grid[0][1] = 'A';
+		grid[0][2] = 'R';
+		grid[1][0] = 'T';
+		grid[1][1] = 'C';
+		grid[1][2] = 'D';
+
+		TreeSet<String> answer = findWords(2,3,grid,new Dictionary());
+
+		Iterator it = answer.iterator();
+
+		while(it.hasNext())
+		{
+			System.out.println(it.next());
+		}
+	}
+
+
+
+	public static TreeSet<String> findWords (int rows, int columns, char[][] grid, Dictionary dictionary)
+	{
+		TreeSet<String> found = new TreeSet<String>();
+
+		//going through to start at each different letter in the grid 
+		for(int i=0; i<rows; i++)			
+		{
+			for(int j = 0; j<columns; j++)
+			{
+				//only considers the starting letter if it is a prefix - if it is not the start of any word in the dictionary, there is no point following it 
+				if(dictionary.isPrefix(Character.toString(grid[i][j])))
+				{
+					checkAround(grid, i, j, dictionary, new boolean[rows][columns], Character.toString(grid[i][j]), found);
+				}
+			}
+		}
+		return found;
+	}
+
+
+	private static void checkAround(char[][] grid, int row, int column, Dictionary dictionary, boolean[][] visited, String word, TreeSet<String> found)
```

Comment:
Nit: I would call "word" "prefix" instead of "word". You don't know it is a word yet.